UM/Controller.py

```diff
@@ -205,10 +205,15 @@ def getTool(self, name: str) -> Optional["Tool"]:
             return None
 
     ##  Get all tools
-    #   \return tools \type{dict}
+    #   \return tools \type{dict} All tools collected in a dictionary via tool name and it's instance
     def getAllTools(self) -> Dict[str, "Tool"]:
         return self._tools
 
+    ##  Set all tools
+    #   \param tools \type{dict} All tools collected in a dictionary via tool name and it's instance
+    def setAllTools(self, tools: dict) -> None:
```

Comment:
Although now typed, I believe the correct typing is `Dict[str, Tool]`.

Reply:
```suggestion
    def setAllTools(self, tools: Dict[str, "Tool"]) -> None:
```

Reply:
Oh, ok. So it would allow different kind of contents inside the dict.
But if we clearly define it as @nallath did for me, then it is clear for the typing that it must be a set of str : "Tool", right?